src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.service.ReportService;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class ReportServiceImplTest {
+    private static final String CATEGORIES = "fruit,quantity";
+    private final ReportService reportService = new ReportServiceImpl();
+
+    @Test
+    void generateReport() {
+        assertEquals(CATEGORIES, reportService.generateReport(Map.of()).get(0));
+        assertEquals("apple,3", reportService.generateReport(Map.of("apple", 3)).get(1));
+        assertEquals("banana,5", reportService.generateReport(Map.of("banana", 5)).get(1));
+        assertEquals("orange,7", reportService.generateReport(Map.of("orange", 7)).get(1));
+    }
+
+    @Test
+    void generateReport_nullMap() {
+        Exception exception = assertThrows(NullPointerException.class,
+                () -> reportService.generateReport(null));
```

Comment:
it's a bit weird to expect and assert NPE, let's change impl in src/main to throw runtime with some message if this happens

Reply:
not fixed, expect illegal argument ex here